docs/MigrationGuides/migration_9_to_10.md

```diff
@@ -0,0 +1,21 @@
+# Migration Guide
+
+This project follows [Semantic Versioning](http://semver.org).
+
+Please follow the appropriate guide below when **upgrading to a new major version** of Moya (e.g. 9.0 -> 10.0).
+
+## Upgrade from 9.x to 10.x
+
+### MoyaError Migration
+- Add `.parameterEncoding`, `.objectMapping`, `.encodableMapping`, cases to `MoyaError` or a `default` case to achieve exhaustiveness.
+
+### Task Migration
+- Add `.requestJSONEncodable` case to `Task` or default case to achieve exhaustiveness.
+
+### Endpoint Migration
+- Replace previously default `method` parameter for `Endpoint.init` with `.get`.
+- Replace previously default `httpHeaderFields` parameter for `Endpoint.init` with `nil`.
+- Replace `urlRequest` with `try? urlRequest()` or use `do/catch` syntax to handle thrown errors.
```

Comment:
We need context for `urlRequest` here (mention somewhere that this is used on `endpoint` or mostly in `endpointClosure`?

Reply:
I have it listed under `Endpoint Migration` so I was hoping that would provide enough context 🤔

Reply:
Oh, I didn't get it then, not so obvious - we can make it similar to what we have in `Changelog`, duplicate the name in the bullet points maybe?

Reply:
I agree. I will make this more clear